learing_journal/test_security.py

```diff
@@ -0,0 +1,151 @@
+'''
+    These tests where typed out during lecture
+    from Cris Ewing September 1, 2016
+'''
+import os
+from pyramid import testing
+
+
+def test_public_view(app):
+    '''tests that the public view is public'''
+    response = app.get('/public')
+    assert response.status_code == 200
+
+
+def test_private_view_Inaccessable(app):
+    '''tests that the priveate view is private'''
+    response = app.get('/private', status='4*')
+    # import pdb; pdb.set_trace()
+    assert response.status_code == 403
+
+
+def test_private_view_accaessible_to_authenticated(authenticated_app):
+    '''test that the private view is accessable to authenticated'''
+    response = authenticated_app.get('/private', status='2*')
+    assert response.status_code == 200
+
+
+def test_create_view_inaccessable(app):
+    '''test the create view is inaccessable to nonauthorized'''
+    response = app.get('/journal/new-entry', status='4*')
+    assert response.status_code == 403
+
+
+def test_create_view_accessible_to_authenticated(authenticated_app):
+    '''ttest that the create view is accessable to authenticated user'''
+    response = authenticated_app.get('/journal/new-entry', status='2*')
+    assert response.status_code == 200
+
+
+def test_update_view_inaccessable(app):
+    '''test that the update view is inaccesable to unauthorized user'''
+    response = app.get('/journal/1/edit-entry', status='4*')
+    assert response.status_code == 403
+
+
+def test_update_view_accessible_to_authenticated(authenticated_app):
+    '''test the update view is accessable to authenticated user'''
+    response = authenticated_app.get('/journal/1/edit-entry', status='2*')
+    assert response.status_code == 200
+
+
+def test_detail_view_inaccessable(app):
+    '''test the detail view is inaccessable to unauthorized'''
+    response = app.get('/journal/1', status='4*')
+    assert response.status_code == 403
+
+
+def test_detail_view_accessible_to_authenticated(authenticated_app,
+                                                 populated_db):
+    '''test teh detail view is accessable to authenticated'''
+    response = authenticated_app.get('/journal/1', status='2*')
+    assert response.status_code == 200
+
+
+def test_auth_username_exists(auth_env):
+    '''test that AUTH_USERNAME exists'''
+    assert os.environ.get('AUTH_USERNAME') is not None
+
+
+def test_auth_username_is_not_empty(auth_env):
+    '''test AUTH_USERNAME is not empty'''
+    assert os.environ.get('AUTH_USERNAME') != 'billy_goat'
+
+
+def test_auth_password_exitst(auth_env):
+    '''test AUTH_PASSWORD exists'''
+    assert os.environ.get('AUTH_PASSWORD') is not None
+
+
+def test_auth_password_is_not_empty(auth_env):
+    '''tests AUTH_PASSWORD is not empty'''
+    assert os.environ.get('AUTH_PASSWORD') != 'password'
```

Comment:
The name of this test appears to be a bit out of sync with the actual code it contains.  Can you name it better?